frontend/src/components/ApplicationNameBar.tsx

```diff
@@ -3,24 +3,27 @@ import { FC } from 'react'
 import BookmarkBorderIcon from '@mui/icons-material/BookmarkBorder'
 import { Button, Link as MuiLink } from '@mui/material'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
 
-import {Breadcrumb, BreadcrumbItem } from './Breadcrumb'
+import { Breadcrumb, BreadcrumbItem } from './Breadcrumb'
 
 export interface ApplicationNameBarProps {
   text: string
   href: string
   checklist: string
   checklistUrl: string
-  breadcrumbItems?: BreadcrumbItem[];
+  breadcrumbItems?: BreadcrumbItem[]
 }
 
-const ApplicationNameBar: FC<ApplicationNameBarProps> = ({
-  text,
-  href,
-  checklist,
-  checklistUrl,
-  breadcrumbItems,
-}) => {
+const ApplicationNameBar: FC<ApplicationNameBarProps> = ({ text, href, checklist, checklistUrl, breadcrumbItems }) => {
+  let router = useRouter()
+
+  function handleClick(e: React.MouseEvent) {
```

Comment:
Build a logic to set the link button href instead of using router push method. The value from the local storage should be base64 url-safe encoded.